src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -567,6 +569,11 @@ void GcsActorManager::DestroyActor(const ActorID &actor_id) {
     RAY_LOG(INFO) << "Tried to destroy actor that does not exist " << actor_id;
     return;
   }
+
+  if (release_resources_) {
```

Comment:
This code is hard to understand outside of the PR. I feel the issue comes from the way we manage the resources. 
Could you please update it to: RayConfig::instance().gcs_actor_scheduling_enabled() at least?

Reply:
Fixed